examples/chip-tool/commands/clusters/ClusterCommand.h

```diff
@@ -148,6 +150,7 @@ class ClusterCommand : public InteractionModelCommands, public ModelCommand, pub
                 clientInfo.peer_node         = mScopedNodeId;
                 clientInfo.monitored_subject = mMonitoredSubject;
                 clientInfo.start_icd_counter = value.ICDCounter;
+                clientInfo.client_type       = mClientType;
```

Comment:
We should not store client type in icd clientInfo, since application knows the client type, and we don't receive any check-in message for ephemeral, i cannot figure out when and how to use clientInfo for which purpose..

Reply:
I am not sure i agree with the statement.
Not persisting the ClientType forces an all or nothing scenario. A client will be only be able to use one type. If it reboots, the only way it knows the type it used at registration if it is not persisted is by always using the same type.

The registration ClientType affects the assumptions a client can do ie. is it garanteed a check-in message will be sent or not and as such we should make sure this information is always available.

> i cannot figure out when and how to use clientInfo for which purpose.

If you do not believe it is necessary, nothing prevents you from not using the information but we shouldn't assume that others might not find a benefit of having this information persisted. It is only a uint8 so the cost is pretty minimal and can enable several use-case we might have missed.

Reply:
Ack, thanks